filesetup.c

```diff
@@ -164,7 +195,8 @@ static int extend_file(struct thread_data *td, struct fio_file *f)
 		else
 			td_verror(td, err, "open");
 		return 1;
-	}
+	} else if (fio_directio(td, f) != 0)
```

Comment:
We had this on td_io_open_fie() before, which is used by anyone wishing to open a file. You moved it to generic_open_file(), which is just one possible implementation that engines can use. This changes the scope of it, have you verified this is actually OK?

Reply:
@axboe I verified it to a limited extent - I tested it did the right thing on macOS (which is was one of the two platforms where it is implemented would affect). How would it have interacted on such platforms previously with engines that implemented their own open that wasn't a regular file?

Reply:
It's the IO engines that will set it, it's not an architecture/platform thing. Grep for ".open_file" in engines/*.c and there are quite a few that do not directly use generic_file_open(). Some of them do call it, so maybe it's fine, but the intent is really for it to be standalone. This means that you cannot put generic code in generic_file_open() that you expect all engines to use.

Reply:
@kusumi got here first in https://github.com/axboe/fio/commit/47534cda051d15767f175967b91ef56771adb0e0 so let's drop this pull request.